### PR TITLE
Faster tensor serialization.

### PR DESCRIPTION
**Pull request by ezyang**
Instead of dynamically allocating a float for each element of the tensor
(lol!) save the tensor itself, and directly read out the data.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torch/csrc/toffee.cpp [hunk 1]
```diff
@@ -23,6 +23,23 @@ bool micropb_encode<double, nullptr>(pb_ostream_t *stream, double* arg) {
...
 
+// TODO: I'm not entirely sure why this can't be in the header...
+bool micropb_callback_tensor(pb_ostream_t *stream, const pb_field_t *field, void * const *arg) {
+  at::Tensor* t = static_cast<at::Tensor*>(*arg);
+  JIT_ASSERT(t->type().scalarType() == at::kFloat);
+  JIT_ASSERT(t->is_contiguous());
+  // TODO: Generalize beyond float
+  // Packed array format!
+  pb_encode_tag(stream, PB_WT_STRING, toffee_TensorProto_float_data_tag);
+  pb_encode_varint(stream, 4 * t->numel()); // number of bytes to write
```
Comment (zdevito) [inline torch/csrc/toffee.cpp:34]: sizeof(float)

**Review by zdevito — APPROVED** [review]

File: torch/csrc/toffee.cpp (hunk 1 above)
Comment (apaszke) [reply torch/csrc/toffee.cpp:34]: There's also a tensor method (element_size) that returns sizeof. You could remove the type assertion in such case.

Comment (apaszke) [reply torch/csrc/toffee.cpp:34]: Also, you can call contiguous and remove the second assert

File: torch/csrc/toffee.cpp [hunk 2]
```diff
@@ -23,6 +23,23 @@ bool micropb_encode<double, nullptr>(pb_ostream_t *stream, double* arg) {
...
+  // TODO: If you have a better way of doing this, I'm all ears.
```
Comment (apaszke) [inline torch/csrc/toffee.cpp:35]: If on element_size and cast to `uintXX_t`

**Review by apaszke — APPROVED** [review]

**Pull request closed without merging**